packages/react/src/components/Checkbox/Checkbox.tsx

```diff
@@ -146,18 +154,28 @@ const Checkbox = React.forwardRef(
         [`${prefix}--checkbox-wrapper--invalid`]: !readOnly && invalid,
         [`${prefix}--checkbox-wrapper--warning`]: showWarning,
         [`${prefix}--checkbox-wrapper--slug`]: slug,
+        [`${prefix}--checkbox-wrapper--decorator`]: decorator,
       }
     );
     const innerLabelClasses = classNames(`${prefix}--checkbox-label-text`, {
       [`${prefix}--visually-hidden`]: hideLabel,
     });
 
-    let normalizedSlug;
-    if (slug && React.isValidElement(slug)) {
-      const size = slug.props?.['kind'] === 'inline' ? 'md' : 'mini';
-      normalizedSlug = React.cloneElement(slug as React.ReactElement<any>, {
-        size,
-      });
+    let normalizedDecorator = slug ?? decorator;
```

Comment:
```suggestion
    let normalizedDecorator = React.isValidElement(slug ?? decorator)
      ? slug ?? decorator
      : null;
```
just rewrote this in my pr's too otherwise storybook throws a minify error when setting the `decorator` prop